core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/util/PropertiesUtils.java

```diff
@@ -421,6 +421,10 @@ public static Object getTypedPropertyFromUri(AbstractCamelModelElement selectedE
 		if (CamelComponentUtils.isTextProperty(p)) {
 			return val;
 		}
+		
+		if (CamelComponentUtils.isClassProperty(p) && !p.getKind().equals(AbstractCamelModelElement.NODE_KIND_ELEMENT) ) {
```

Comment:
Optional: CamelComponentUtils.isClassProperty(p) && !p.getKind().equals(AbstractCamelModelElement.NODE_KIND_ELEMENT) is currently used in 3 places.
We can consider to create a specific method in CamelComponentUtils